### DISCUSSION TITLE
How to convert from decimal to hex?

### DISCUSSION THREAD

**Discussion by FoolMuun** [schibsted/jslt#304] [Q&A] [answered]
I'm stumped. How does one go about converting decimal values to hex values using JSLT?
E.g. Hex(1234)="4D2"

**Comment by FoolMuun** [comment, marked as answer]
Here's a possible solution:
```
def hex(dec)
  if ($dec==0) 0
  else hexcalc($dec,"")

def hexcalc(dec, result)
  let hexdict = ["0","1","2","3","4","5","6","7","8","9","A","B","C","D","E","F"]
  if ($dec!=0) hexcalc(floor($dec/16), $hexdict[mod($dec,16)] + $result)  
  else $result

hex(1234)
```
Output:
```"4D2"```

**Reply by FoolMuun** [reply]
Not safe for negative values though, but that can be fixed.

**Comment by larsga** [comment]
Great to see that you managed to sort this out yourself, and thank you for posting the solution so other people can use it. I think if this question shows up again we'll seriously consider adding the function to the language.
